### ISSUE TITLE
large photo causing exception

### ISSUE THREAD

**Issue by ekilah** [contactpush/contact#29]
I took a photo and set it as my photo, viewing that profile from another phone causes this @rtteal (given the debug msg about size I think you're expecting issues still with this):

03-16 00:41:44.035    2874-2874/com.codepath.contact D/CreateProfileFragment﹕ photo is 1973545 bytes
03-16 00:41:44.067    2874-2874/com.codepath.contact I/art﹕ Alloc partial concurrent mark sweep GC freed 2395(122KB) AllocSpace objects, 5(5MB) LOS objects, 18% free, 72MB/88MB, paused 468us total 28.584ms
03-16 00:41:44.093    2874-2874/com.codepath.contact I/art﹕ Alloc concurrent mark sweep GC freed 1275(93KB) AllocSpace objects, 0(0B) LOS objects, 18% free, 72MB/88MB, paused 306us total 23.727ms
03-16 00:41:44.093    2874-2874/com.codepath.contact I/art﹕ Forcing collection of SoftReferences for 36MB allocation
03-16 00:41:44.123    2874-2874/com.codepath.contact I/art﹕ Alloc concurrent mark sweep GC freed 15(488B) AllocSpace objects, 0(0B) LOS objects, 18% free, 72MB/88MB, paused 921us total 30.307ms
03-16 00:41:44.124    2874-2874/com.codepath.contact E/art﹕ Throwing OutOfMemoryError "Failed to allocate a 38340876 byte allocation with 16777216 free bytes and 23MB until OOM"
03-16 00:41:44.124    2874-2889/com.codepath.contact I/art﹕ WaitForGcToComplete blocked for 84.041ms for cause Background
03-16 00:41:44.124    2874-2874/com.codepath.contact D/skia﹕ --- decoder->decode returned false
03-16 00:41:44.124    2874-2874/com.codepath.contact D/AndroidRuntime﹕ Shutting down VM
03-16 00:41:44.124    2874-2874/com.codepath.contact E/CrashReporting﹕ ParseCrashReporting caught a OutOfMemoryError exception for com.codepath.contact. Building report.
03-16 00:41:44.125    2874-2874/com.codepath.contact E/CrashReporting﹕ Handling exception for crash
    java.lang.OutOfMemoryError: Failed to allocate a 38340876 byte allocation with 16777216 free bytes and 23MB until OOM
            at dalvik.system.VMRuntime.newNonMovableArray(Native Method)
            at android.graphics.BitmapFactory.nativeDecodeByteArray(Native Method)
            at android.graphics.BitmapFactory.decodeByteArray(BitmapFactory.java:522)
            at android.graphics.BitmapFactory.decodeByteArray(BitmapFactory.java:545)
            at com.codepath.contact.fragments.CreateProfileFragment.setProfileImage(CreateProfileFragment.java:645)
            at com.codepath.contact.fragments.CreateProfileFragment.setCurrentValues(CreateProfileFragment.java:456)
            at com.codepath.contact.fragments.CreateProfileFragment.access$500(CreateProfileFragment.java:49)
            at com.codepath.contact.fragments.CreateProfileFragment$7.receiveContact(CreateProfileFragment.java:438)
            at com.codepath.contact.models.ContactInfo$2.done(ContactInfo.java:56)
            at com.codepath.contact.models.ContactInfo$2.done(ContactInfo.java:52)
            at com.parse.Parse$5$1.run(Parse.java:924)
            at android.os.Handler.handleCallback(Handler.java:739)
            at android.os.Handler.dispatchMessage(Handler.java:95)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5221)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
03-16 00:41:44.125    2874-2874/com.codepath.contact D/CrashReporting﹕ Generating report file for crash
03-16 00:41:44.145    2874-2874/com.codepath.contact D/CrashReporting﹕ Retrieving logcat output...
03-16 00:41:44.154    2874-2874/com.codepath.contact D/CrashReporting﹕ Retrieving logcat output...
03-16 00:41:44.166    2874-2874/com.codepath.contact D/CrashReporting﹕ Retrieving logcat output...
03-16 00:41:44.187    2874-2874/com.codepath.contact W/Settings﹕ Setting stay_on_always has moved from android.provider.Settings.System to android.provider.Settings.Global, returning read-only value.
03-16 00:41:44.224    2874-2874/com.codepath.contact V/CrashReporting﹕ ParseCrashReporting caught an OutOfMemoryError. Report upload deferred until next ParseCrashReporting launch.
    --------- beginning of crash
03-16 00:41:44.224    2874-2874/com.codepath.contact E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: com.codepath.contact, PID: 2874
    java.lang.OutOfMemoryError: Failed to allocate a 38340876 byte allocation with 16777216 free bytes and 23MB until OOM
            at dalvik.system.VMRuntime.newNonMovableArray(Native Method)
            at android.graphics.BitmapFactory.nativeDecodeByteArray(Native Method)
            at android.graphics.BitmapFactory.decodeByteArray(BitmapFactory.java:522)
            at android.graphics.BitmapFactory.decodeByteArray(BitmapFactory.java:545)
            at com.codepath.contact.fragments.CreateProfileFragment.setProfileImage(CreateProfileFragment.java:645)
            at com.codepath.contact.fragments.CreateProfileFragment.setCurrentValues(CreateProfileFragment.java:456)
            at com.codepath.contact.fragments.CreateProfileFragment.access$500(CreateProfileFragment.java:49)
            at com.codepath.contact.fragments.CreateProfileFragment$7.receiveContact(CreateProfileFragment.java:438)
            at com.codepath.contact.models.ContactInfo$2.done(ContactInfo.java:56)
            at com.codepath.contact.models.ContactInfo$2.done(ContactInfo.java:52)
            at com.parse.Parse$5$1.run(Parse.java:924)
            at android.os.Handler.handleCallback(Handler.java:739)
            at android.os.Handler.dispatchMessage(Handler.java:95)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5221)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)

**Comment by ekilah** [comment]
I deleted my photo on parse for now to unblock btw

**Comment by rtteal** [comment]
I think this is fixed now.  I can't really test this too well on the emulator.  Can you test and let me know if the problem is resolved?  Also, let me know if the image taken from the camera is rotated the right way.

**Comment by ekilah** [comment]
will try tomorrow.

**Comment by ekilah** [comment]
The picture taken from the camera is actually saved to parse rotated 180 degrees. @rtteal

**Comment by ekilah** [comment]
made a new issue (#41) for the rotation issue.
